br.gov.fazenda.receita.irpf2023.yaml

```diff
@@ -0,0 +1,229 @@
+id: br.gov.fazenda.receita.irpf2023
+runtime: org.gnome.Platform
+runtime-version: '44'
+sdk: org.gnome.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.openjdk11
+tags:
+  - proprietary
+command: irpf2023
+
+finish-args:
+  - --share=network
+  - --socket=x11
+  - --socket=cups # Printing doesn't really work righ now: https://github.com/flatpak/flatpak/issues/3355
+  - --share=ipc
+  - --env=PATH=/app/jre/bin:/app/bin:/usr/bin
+  - --env=JAVA_HOME=/app/jre
+  # The next two permissions (plus the GNOME runtime) are needed to make most clickable URLs within the app open correctly.
+  - --talk-name=org.gtk.vfs.*
+  - --filesystem=xdg-run/gvfsd
+  - --filesystem=xdg-documents # Useful place to save PDF files.
+  - --filesystem=~/ProgramasRFB:create # This path is hardcoded by the app to save all kinds of tax data.
+  - --persist=.java # In-app preferences are stored in: ~/.java/.userPrefs/serpro
+  - --persist=.receitanet # Log files, tokens and whatnot.
+  - --persist=.rfb # User-provided form data (such as CPF and CNPJ numbers) are stored in: ~/.rfb
+
+modules:
+  - name: irpf2023
+    buildsystem: simple
+    build-commands:
+      - install -Dm755 -t /app/bin irpf2023
+      - install -Dm644 -t /app/share/applications $FLATPAK_ID.desktop
+      - install -Dm644 icon-64.png /app/share/icons/hicolor/64x64/apps/$FLATPAK_ID.png
+      - install -Dm644 icon-128.png /app/share/icons/hicolor/128x128/apps/$FLATPAK_ID.png
+      - install -Dm644 icon-256.png /app/share/icons/hicolor/256x256/apps/$FLATPAK_ID.png
+      - install -Dm644 -t /app/share/metainfo $FLATPAK_ID.metainfo.xml
+      - install -Dm644 -t /app/share/$FLATPAK_ID irpf.jar pgd-updater.jar
+      - cp -r --no-preserve=mode help/ lib/ lib-modulos/ /app/share/$FLATPAK_ID
+      # Workaround to make the clickable link on the "About" window open with the default web browser.
+      - ln -s /usr/bin/xdg-open /app/bin/firefox
+    sources:
+      - type: archive
+        url: https://downloadirpf.receita.fazenda.gov.br/irpf/2023/irpf/arquivos/IRPF2023-1.2.zip
+        sha256: d24ae893e0e3e5f680b85b0be5a663053c47af0d8148406732636fdedb8ad65e
+        x-checker-data:
+          type: html
+          url: https://www.gov.br/receitafederal/pt-br/centrais-de-conteudo/download/pgd/dirpf
+          version-pattern: /arquivos/IRPF2023-([\d.]+)\.zip
+          url-template: https://downloadirpf.receita.fazenda.gov.br/irpf/2023/irpf/arquivos/IRPF2023-$version.zip
+          is-main-source: true
+      - type: script
+        dest-filename: irpf2023
+        commands:
+          - exec java -jar /app/share/$FLATPAK_ID/irpf.jar $@
```

Comment:
This works, but am I violating some best practices here?

I'm trying really hard to avoid typing `br.gov.fazenda.receita.irpf2023` everywhere, just to make packaging easier for next years. ^^

Reply:
It's fine everywhere else but here. I don't know how robust this is during run time.

Reply:
Thanks! Fixed it.